go-controller/pkg/util/net_linux.go

```diff
@@ -238,8 +238,8 @@ func LinkAddrExist(link netlink.Link, address *net.IPNet) (bool, error) {
 }
 
 // LinkAddrAdd removes existing addresses on the link and adds the new address
```

Comment:
probably should add a comment here that if both preferred and valid lifetime is 0, then they are not applied but if either are greater than 0, then both are applied.